error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJDurationRules.java

```diff
@@ -0,0 +1,119 @@
+package tech.picnic.errorprone.refasterrules;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import java.time.Duration;
+import org.assertj.core.api.AbstractBooleanAssert;
+import org.assertj.core.api.AbstractDurationAssert;
+import org.assertj.core.api.AbstractLongAssert;
+import tech.picnic.errorprone.refaster.annotation.OnlineDocumentation;
+
+/**
+ * Refaster rules related to AssertJ assertions over {@link Duration}s.
+ *
+ * <p>These rules simplify and improve the readability of Duration assertions by using the more
+ * specific AssertJ Duration assertion methods instead of generic assertions on the Duration's
+ * numeric values.
+ */
+@OnlineDocumentation
+final class AssertJDurationRules {
+  private AssertJDurationRules() {}
+
+  static final class AbstractDurationAssertHasNanos {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long nanos) {
+      return assertThat(duration.toNanos()).isEqualTo(nanos);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long nanos) {
+      return assertThat(duration).hasNanos(nanos);
+    }
+  }
+
+  static final class AbstractDurationAssertHasMillis {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long millis) {
+      return assertThat(duration.toMillis()).isEqualTo(millis);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long millis) {
+      return assertThat(duration).hasMillis(millis);
+    }
+  }
+
+  static final class AbstractDurationAssertHasSeconds {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long seconds) {
+      return assertThat(duration.toSeconds()).isEqualTo(seconds);
```

Comment:
We can also rewrite `getSeconds()` calls. (Further up, that's not true for `getNanos()`.)

_Edit:_ no need; [`JavaDurationGetSecondsToToSeconds`](https://errorprone.info/bugpattern/JavaDurationGetSecondsToToSeconds) rewrites `getSeconds()` to `toSeconds()` already.